sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -290,7 +300,9 @@ case class HiveScriptIOSchema (
     outputSerdeClass: Option[String],
     inputSerdeProps: Seq[(String, String)],
     outputSerdeProps: Seq[(String, String)],
-    schemaLess: Boolean) extends ScriptInputOutputSchema with HiveInspectors {
+    schemaLess: Boolean,
+    recordWriter: String,
+    recordReader: String) extends ScriptInputOutputSchema with HiveInspectors {
```

Comment:
Use `Option[String]` instead of `String` for `recordWriter` and `recordReader`, and don't use empty string as their default value in other places of this PR.